fenicsxprecice/expression_core.py

```diff
@@ -55,11 +56,11 @@ def update_boundary_data(self, vals, coords_x, coords_y=None, coords_z=None):
         self._f = self.create_interpolant()
 
         if self.is_scalar_valued():
-            assert (self._vals.shape == self._coords_x.shape)
+            assert (self._vals.shape[0] == self._coords_x.shape[0])
         elif self.is_vector_valued():
             assert (self._vals.shape[0] == self._coords_x.shape[0])
 
-    def interpolate(self, x):
+    def interpolate_precice(self, x):
```

Comment:
Why was this renamed? If you can use the old name, I would suggest to keep it to reduce the diff.

Reply:
I think I know what's the problem now: `dolfinx.fem.Function` already has a function `dolfinx.fem.Function.interpolate`. This function is probably needed by the class. Therefore, we are not allowed to override it. I'm still not a big fan of the name `interpolate_precice`. Maybe we can get rid of the function `interpolate_precice` entirely and put everything into `update_boundary_data`. I will try to get this working.

Reply:
I managed to simplify the implementation in c8e4b75a9ae71104ffef81e662e458149cb7a0dd. It was quite convoluted before and now it is much nicer (because of the way how FEniCS-X allows us to create an interpolant). @PhilipHildebrand: Please take a look at c8e4b75a9ae71104ffef81e662e458149cb7a0dd. We can also discuss the changes together.